### PR TITLE
New dataset scripts: morphoMNIST, Shapes3D, Cars3D

### PR DESCRIPTION
**Pull request by AnejSvete**
Added scripts for downloading and managing 3 additional datasets:
- `morphoMNIST`
- `Cars3D`
- `Shapes3D`

All are based on the same idea as the script for `dSprites` and share some utility code.

**EDIT**:
Implemented the comments, now with Temporary files, a few convenience functions, and better docstrings.

### CODE REVIEW

File: latte/dataset/cars3d.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    """Function extracting and preparing the Cars3D data.
+
+    Args:
+        filepath: directory for storing the temporary data and the dataset
+
+    Returns:
+        Cars3DDataset object
+    """
+
+    archive_filepath = filepath / "cars3d.tar.gz"
```
Comment (pawel-czyz) [inline latte/dataset/cars3d.py:41]: As described below in more detail, in the future you may consider the https://docs.python.org/3/library/tempfile.html :slightly_smiling_face:

**Review by pawel-czyz — APPROVED** [review]
Hi Anej,

Thank you a lot for this PR! I have added some suggestions, but I don't think they are any obstacle to merging it now.

File: latte/dataset/morphoMNIST.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    train_images = dsutils.load_idx(str(extract_filepath / os.path.join(version, "train-images-idx3-ubyte.gz")))
+    train_labels = dsutils.load_idx(str(extract_filepath / os.path.join(version, "train-labels-idx1-ubyte.gz")))
+    train_latents = pd.read_csv(str(extract_filepath / os.path.join(version, "train-morpho.csv")))
+
+    test_images = dsutils.load_idx(str(extract_filepath / os.path.join(version, "t10k-images-idx3-ubyte.gz")))
+    test_labels = dsutils.load_idx(str(extract_filepath / os.path.join(version, "t10k-labels-idx1-ubyte.gz")))
+    test_latents = pd.read_csv(str(extract_filepath / os.path.join(version, "t10k-morpho.csv")))
+
+    train_perturbations, test_perturbations = None, None
+    if version in ["local", "global"]:
```
[the remark this answers is not in the document]
Comment (AnejSvete) [reply latte/dataset/morphoMNIST.py:113]: Nothing. This data is not there for "plain", so the field in the object should be empty. Should this be handled differently?

File: latte/dataset/morphoMNIST.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        filepath: directory where the dataset should be stored
+        version: the version of the data to load (plain or local/global perturbations)
+        _source: the URL of the data to be downloaded
+        _md5_checksum: the MD5 checksum of the file to be downloaded
+
+    Returns:
+        MorphoMNISTDataset object
+    """
+    archive_filepath = filepath / f"{version}_archive.zip"
+    extract_filepath = filepath
```
[the remark this answers is not in the document]
Comment (AnejSvete) [reply latte/dataset/morphoMNIST.py:93]: Yes! I've never used this, thanks! I'll implement it, I think it'll be way cleaner :)

[the remark this answers is not in the document]
Comment (pawel-czyz) [reply latte/dataset/morphoMNIST.py:93]: Whatever way you prefer – the code already looks readable to me :wink: Just in the future you may prefer this one.

**Review by pawel-czyz — APPROVED** [review]
Excellent!

**Pull request merged**